src/crypto/clu_crypto_setup.c

```diff
@@ -0,0 +1,410 @@
+                char* keyString;
+
+                keyString = (char*)XMALLOC(XSTRLEN(optarg), HEAP_HINT,
+                        DYNAMIC_TYPE_TMP_BUFFER);
+                if (keyString == NULL) {
+                    wolfCLU_freeBins(pwdKey, iv, key, NULL, NULL);
+                    return MEMORY_E;
+                }
+
+                XSTRNCPY(keyString, optarg, XSTRLEN(optarg));
+                ret = wolfCLU_hexToBin(keyString, &key, &numBits,
+                                       NULL, NULL, NULL,
+                                       NULL, NULL, NULL,
+                                       NULL, NULL, NULL);
+                XFREE(keyString, HEAP_HINT, DYNAMIC_TYPE_TMP_BUFFER);
+                if (ret != 0) {
+                    WOLFCLU_LOG(WOLFCLU_L0,
+                            "failed during conversion of Key, ret = %d", ret);
+                    wolfCLU_freeBins(pwdKey, iv, key, NULL, NULL);
+                    return WOLFCLU_FATAL_ERROR;
+                }
+                keyCheck = 1;
+                keyType = 2;
+            }
+            break;
+
+        case WOLFCLU_SIGFILE:
+            break;
+
+        case WOLFCLU_DECRYPT:
+            encCheck = 0;
+            decCheck = 1;
+            break;
+
+        case WOLFCLU_DEBUG:
+            verbose = 1;
+            break;
+
+        case WOLFCLU_MD:
+            hashType = wolfSSL_EVP_get_digestbyname(optarg);
+            if (hashType == NULL) {
+                WOLFCLU_LOG(WOLFCLU_L0, "invalid digest name");
+                return WOLFCLU_FATAL_ERROR;
+            }
+            break;
+
+        case ':':
+        case '?':
+            break;
+
+        default:
+            /* do nothing. */
+            (void)ret;
+        }
+    }
+
+    if (pwdKeyChk == 0 && keyCheck == 0) {
+        if (decCheck == 1) {
+            WOLFCLU_LOG(WOLFCLU_L0, "\nDECRYPT ERROR:");
+            WOLFCLU_LOG(WOLFCLU_L0, "no key set or passphrase");
+            WOLFCLU_LOG(WOLFCLU_L0,
+                    "Please type \"wolfssl -decrypt -help\" for decryption"
+                                                            " usage \n");
+            return WOLFCLU_FATAL_ERROR;
+        }
+        /* if no pwdKey is provided */
+        else {
+            WOLFCLU_LOG(WOLFCLU_L0,
+                    "No -pwd flag set, please enter a password to use for"
+                    " encrypting.");
+            WOLFCLU_LOG(WOLFCLU_L0,
+                    "Write your password down so you don't forget it.");
+            ret = wolfCLU_noEcho((char*)pwdKey, keySize);
+            pwdKeyChk = 1;
+        }
+    }
+
+    if (inCheck == 0 && encCheck == 1) {
+        ret = 0;
+        while (ret == 0) {
+            WOLFCLU_LOG(WOLFCLU_L0,
+                    "-in flag was not set, please enter a string or"
+                   "file name to be encrypted: ");
+            ret = (int) scanf("%s", inName);
+        }
+        in = inName;
+        /* if no input is provided */
```

Comment:
I don't get what this comment means. Input **was** provided in the loop above if we made it here.